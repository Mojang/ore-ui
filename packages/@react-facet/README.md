# @react-facet

<img src="logo.png" width="640px" />

Observable-based state management for performant game UIs built in React

- Read more in the [Documentation Site](https://react-facet.mojang.com/)
