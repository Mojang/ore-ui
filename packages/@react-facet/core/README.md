# @react-facet/core

Documentation coming soon.
