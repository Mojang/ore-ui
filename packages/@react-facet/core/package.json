{
  "name": "@react-facet/core",
  "repository": {
    "type": "git",
    "url": "https://github.com/Mojang/ore-ui.git",
    "directory": "packages/@react-facet/core"
  },
  "author": {
    "name": "Paulo Ragonha",
    "email": "paulo@ragonha.me"
  },
  "contributors": [
    {
      "name": "Danila Dergachev",
      "email": "dderg@protonmail.com"
    },
    {
      "name": "Fernando Via Canel",
      "email": "fernando.via@gmail.com"
    },
    {
      "name": "Marlon Huber-Smith",
      "email": "marlonhubersmith@gmail.com"
    }
  ],
  "files": [
    "dist/**/*"
  ],
  "homepage": "https://react-facet.mojang.com/",
  "bugs": "https://github.com/Mojang/ore-ui/issues",
  "license": "MIT",
  "version": "0.3.0",
  "main": "src/index.ts",
  "publishConfig": {
    "main": "dist/index.js",
    "types": "dist/index.d.ts"
  },
  "sideEffects": false,
  "scripts": {
    "build": "rimraf dist && tsc",
    "prepublish": "yarn build"
  },
  "peerDependencies": {
    "react": "^16.9.0"
  },
  "devDependencies": {
    "@react-facet/dom-fiber-testing-library": "0.3.0",
    "@types/react": "^17.0.11",
    "@types/react-reconciler": "^0.18.0",
    "@types/rimraf": "^3",
    "cross-env": "^7.0.3",
    "react": "^16.9.0",
    "rimraf": "^3.0.2",
    "typescript": "^4.3.5"
  }
}
