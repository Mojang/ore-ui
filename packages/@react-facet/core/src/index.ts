export * from './components'
export * from './createEqualityChecks'
export * from './createFacetContext'
export * from './equalityChecks'
export * from './facet'
export * from './helpers'
export * from './hooks'
export * from './mapFacets'
export * from './types'
