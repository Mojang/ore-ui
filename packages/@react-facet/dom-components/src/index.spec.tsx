import React from 'react'
import { render, act } from '@testing-library/react'
import { fast } from '.'
import { createFacet } from '@react-facet/core'

it('renders several fast components properly', () => {
  const idFacet = createFacet({ initialValue: 'some-id-name' })
  const classNameFacet = createFacet({ initialValue: 'some-class-name' })
  const textFacet = createFacet({ initialValue: 'lorem ipsum' })
  const imgSrcFacet = createFacet({ initialValue: 'about:blank' })
  const valueFacet = createFacet({ initialValue: 'placeholder' })
  const backgroundColorFacet = createFacet({ initialValue: 'red' })
  const numberFacet = createFacet({ initialValue: 1 })

  const Example = () => {
    return (
      <fast.div id={idFacet} className={classNameFacet} style={{ backgroundColor: backgroundColorFacet }}>
        <fast.p id={idFacet} className={classNameFacet} style={{ backgroundColor: backgroundColorFacet }}>
          <fast.span id={idFacet} className={classNameFacet} style={{ backgroundColor: backgroundColorFacet }}>
            <fast.text id={idFacet} text={textFacet} />
            <fast.text text={numberFacet} />
          </fast.span>
        </fast.p>

        <fast.input
          id={idFacet}
          className={classNameFacet}
          value={valueFacet}
          style={{ backgroundColor: backgroundColorFacet }}
        />
        <fast.textarea
          id={idFacet}
          className={classNameFacet}
          value={valueFacet}
          style={{ backgroundColor: backgroundColorFacet }}
        />

        <fast.img
          id={idFacet}
          className={classNameFacet}
          src={imgSrcFacet}
          style={{ backgroundColor: backgroundColorFacet }}
        />
      </fast.div>
    )
  }

  const scenario = <Example />

  const { container } = render(scenario)

  expect(container.firstChild).toMatchSnapshot()
})

it('does not trigger onMouseEnter on render', () => {
  const mock = jest.fn()
  const Example = () => {
    return <fast.div onMouseEnter={mock}></fast.div>
  }
  act(() => {
    render(<Example />)
  })
  expect(mock).toHaveBeenCalledTimes(0)
})

it('does not trigger onMouseLeave on render', () => {
  const mock = jest.fn()
  const Example = () => {
    return <fast.div onMouseLeave={mock}></fast.div>
  }
  act(() => {
    render(<Example />)
  })
  expect(mock).toHaveBeenCalledTimes(0)
})
