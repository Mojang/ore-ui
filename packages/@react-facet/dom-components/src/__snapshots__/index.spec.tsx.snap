// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders several fast components properly 1`] = `
<div
  class="some-class-name"
  id="some-id-name"
  style="background-color: red;"
>
  <p
    class="some-class-name"
    id="some-id-name"
    style="background-color: red;"
  >
    <span
      class="some-class-name"
      id="some-id-name"
      style="background-color: red;"
    >
      <span
        class=""
        id="some-id-name"
      >
        lorem ipsum
      </span>
      <span
        class=""
        id=""
      >
        1
      </span>
    </span>
  </p>
  <input
    class="some-class-name"
    id="some-id-name"
    style="background-color: red;"
    value="placeholder"
  />
  <textarea
    class="some-class-name"
    disabled=""
    id="some-id-name"
    style="background-color: red;"
    value="placeholder"
  />
  <img
    class="some-class-name"
    id="some-id-name"
    src="about:blank"
    style="background-color: red;"
  />
</div>
`;
